scripts/tests/run_test_suite.py

```diff
@@ -160,7 +160,7 @@ def main(context, dry_run, log_level, target, target_glob, target_skip_glob,
     # Figures out selected test that match the given name(s)
     if runtime == TestRunTime.CHIP_REPL_PYTHON:
         all_tests = [test for test in chiptest.AllReplYamlTests()]
-    elif runtime == TestRunTime.CHIP_TOOL_PYTHON:
+    elif runtime == TestRunTime.CHIP_TOOL_PYTHON and os.path.basename(chip_tool) != "darwin-framework-tool":
```

Comment:
This feels unclean. How about another constant of DARWIN_TOOL_PYTHON?

Reply:
Unsure what the blast radius of that is... I checked how often CHIP_TOOL_PYTHON is used and it does not seem to be used that much (just 3 times) and I imagine some logic needed for `chip_tool_with_python`.

If that is really impossible then maybe just some override on what tests to run could be used. the extra basename check and fallback into "all chip tool tests" seems not ideal.

Reply:
I have some other patches that i would push once this one lands. Everything is in https://github.com/vivien-apple/connectedhomeip-1/tree/MattterYamlTests_RemoveCodegenTests and is mostly ready to be made as PRs.
In https://github.com/vivien-apple/connectedhomeip-1/commit/e747328dffd87a636bcc263551c4267ab803eac3 I am updating the TestRuntime enum to be cleaner! 
Would you be fine with a followup ?

Reply:
Ok, I take that this proves that this is indeed getting better in time, am happy with that.